### PR TITLE
Use dblink to send SQL requests

### PR DESCRIPTION
**Pull request by tbille**
# Summary

## The problem 

Due to the lack of access that we can get from our K8s cluster to the redshift database we need to find alternative solutions to access it.

One solution is to give access to redshift to our own database a run the query from it. This is [documented in this article](https://aws.amazon.com/blogs/big-data/join-amazon-redshift-and-amazon-rds-postgresql-with-dblink/) from AWS.

This PR refactors the code to now run all queries through an external database. The database can be local or distant. Until [this ticket](https://portal.admin.canonical.com/C133921/) is not done we will need to use a local database for dev.

Fixes https://github.com/canonical-web-and-design/workplace-engineering-squad/issues/157

## Done

- Refactor code to run all SQL queries through an external database
- Add setup and reset scripts
- Update Readme

# QA

- Follow the steps in the README
- From those steps you should be able to setup your database and start the service
- Once it started, with multiple users access pages like: http://0.0.0.0:8501/stages/application-review
- Make sure the data makes sense :smile:

### CODE REVIEW

**Review by andesol — APPROVED** [review]
Works fine for me! One difference I noticed is that we used to be able to see the data when visiting localhost:8501/stages/application-review and similar, while now only 0.0.0.0:8501 works.

**Comment by tbille** [comment]
> Works fine for me! One difference I noticed is that we used to be able to see the data when visiting `localhost:8501/stages/application-review` and similar, while now only `0.0.0.0:8501` seems to work.

@andesol  Can you give more details? Your database can't access the data? Did you follow the steps in the readme for the local database?

File: konf/site.yaml [hunk 1]
```diff
@@ -6,9 +6,9 @@ env:
   - name: SENTRY_DSN
     value: https://f51f18282e7d44bebe6cc268ba96f7a2@sentry.is.canonical.com//49
   
-  - name: GREENHOUSE_DATABASE_URL
+  - name: WORKPLACE_ENGINEERING_DATABASE_URL
     secretKeyRef:
-      key: greenhouse_database_url
+      key: workplace_egineering_database_url
```
[the remark this answers is not in the document]
Comment (tbille) [reply konf/site.yaml]: Good catch, fixed

File: webapp/stages/shared/services.py [hunk 2]
```diff
@@ -12,24 +12,19 @@
...
+        FROM jobs j
+            JOIN hiring_team ht on j.id = ht.job_id
+            JOIN users u on ht.user_id = u.id
+            JOIN departments d on j.department_id = d.id
+        WHERE ht.role = 'Recruiter'
+            AND ht.responsible=true
+            AND u.email='{user_email}'
+            AND j.status='open'
+        """,
+        ["job_id int8", "job_name varchar", "departement_name varchar"],
```
[the remark this answers is not in the document]
Comment (tbille) [reply webapp/stages/shared/services.py]: Good catch, fixed

**Comment by andesol** [comment]
@tbille Forget about it. I thought that http://0.0.0.0:8501/stages/application-review/performance.json (for example) was working but the same with `localhost` (http://localhost:8501/stages/application-review/performance.json) wasn't. But I double checked and it was likely something I didn't get right.

It works, let's merge!

**Pull request merged**